src/components/FindMovie/FindMovie.js

```diff
@@ -1,55 +1,119 @@
-          />
-        </div>
+import getMovieFromServer from '../../api';
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+export const FindMovie = ({ addMovie, movies }) => {
+  const [movie, setMovie] = useState(null);
+  const [Title, setTitle] = useState('');
+  const [hasMovie, setHasMovie] = useState(false);
+  const [cantFind, setCantFind] = useState(false);
+
+  const handleFindMovie = async() => {
+    const newMovie = await getMovieFromServer(Title);
+
+    // eslint-disable-next-line
+    if (newMovie.hasOwnProperty('Error')) {
+      setHasMovie(false);
+      setCantFind(true);
+
+      return;
+    }
+
+    setCantFind(false);
+    setHasMovie(true);
+    setMovie(newMovie);
+  };
+
+  const handleAddMovie = () => {
+    if (!movie) {
+      return;
+    }
+
+    if (movies.some(film => film.imdbID === movie.imdbID)) {
+      return;
+    }
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+    addMovie(movie);
+    setTitle('');
+    setMovie(null);
+    setHasMovie(false);
+  };
+
+  return (
+    <>
+      <form
+        className="find-movie"
+        onSubmit={(event) => {
+          event.preventDefault();
+        }}
+      >
+        <div className="field">
+          <label className="label" htmlFor="movie-Title">
+            Movie Title
+          </label>
+
+          <div className="control">
+            <input
+              type="text"
+              id="movie-Title"
+              placeholder="Enter a Title to search"
+              className={`input ${cantFind && 'is-danger'}`}
+              value={Title}
+              onChange={(event) => {
+                setTitle(event.target.value);
+              }}
+            />
+          </div>
+
+          {cantFind && (
+          <p className="help is-danger">
+            Can&apos;t find a movie with such a Title
+          </p>
+          )}
         </div>
 
-        <div className="control">
-          <button
-            type="button"
-            className="button is-primary"
-          >
-            Add to the list
-          </button>
+        <div className="field is-grouped">
+          <div className="control">
+            <button
+              type="button"
+              className="button is-light"
+              onClick={handleFindMovie}
+            >
+              Find a movie
+            </button>
+          </div>
+
+          <div className="control">
+            <button
+              type="button"
+              className="button is-primary"
+              onClick={handleAddMovie}
+            >
+              Add to the list
+            </button>
+          </div>
         </div>
+      </form>
+
+      <div className="container">
+
+        {hasMovie
```

Comment:
```suggestion
        {movie
```
Actually, you don't need this variable. For conditional rendering, you can use the `movie` variable. If there's no movie, it will be null and nothing will render